File: Lokalano-partnerstvo/Core/Entities/Partner.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Core.Entities
{
    public class Partner : BaseEntity
    {
        public string UserId { get; set; }
        public string Ime { get; set; }
        public string Opis { get; set; }
        public string Telefon { get; set; }
        public string Fax { get; set; }
        public string Mail { get; set; }
        public string Adresa { get; set; }
        public string Web { get; set; }
        public string Facebook { get; set; }
        public string Instagram { get; set; }
        public string Tiktok { get; set; }
        public string Youtube { get; set; }
        public string Twitter { get; set; }
        public string  ImageUrl { get; set; }
        public Photo Photo { get; set; }
        public int? PhotoId { get; set; }

        public void AddPhoto(string imageUrl, string fileName)
        {
            var photo = new Photo
            {
                FileName = fileName,
                PictureUrl = imageUrl
            };

            Photo = photo;
        }
        public void RemovePhoto()
        {
            if (Photo != null)
            {
                Photo = null;
            }
        }
    }
}

File: Lokalano-partnerstvo/client/src/app/shared/models/partner.ts
export interface Partner {
  id: number;
  userId: string;
  ime: string;
  opis: string;
  telefon: string;
  fax: string;
  mail: string;
  adresa: string;
  web: string;
  facebook: string;
  instagram: string;
  tiktok: string;
  youtube: string;
  twitter: string;
  imageUrl: string;
  photo: Photo;
}

export interface Photo {
  id: number;
  imageUrl: string;
  fileName: string;
}

export interface PartnerToCreate {
  userId: string;
  ime: string;
  opis: string;
  telefon: string;
  fax: string;
  mail: string;
  adresa: string;
  web: string;
  facebook: string;
  instagram: string;
  tiktok: string;
  youtube: string;
  twitter: string;
  imageUrl: string;
}

export interface PartnerToUpdate {
  opis: string;
  telefon: string;
  fax: string;
  adresa: string;
  web: string;
  facebook: string;
  instagram: string;
  tiktok: string;
  youtube: string;
  twitter: string;
  imageUrl: string;
}

export class PartnerFormValues implements PartnerToCreate {
  userId = '';
  ime = '';
  opis = '';
  telefon = '';
  fax = '';
  mail = '';
  adresa = '';
  web = '';
  facebook = '';
  instagram = '';
  tiktok = '';
  youtube = '';
  twitter = '';
  imageUrl = '';

  constructor(init?: PartnerFormValues) {
    Object.assign(this, init);
  }

}
